app/models/user.rb

```diff
@@ -3,8 +3,51 @@ class User < ActiveRecord::Base
   # validates password and password_confirmation; no validates_presence_of necessary!!!
   has_secure_password validations: true
   has_many :queue_items, -> { order('order_value') }
+  has_many :reviews
 
   validates_presence_of :email, :full_name
   validates_uniqueness_of :email
 
+  def update_queue_items(queue_items_hash)
+    ActiveRecord::Base.transaction do
+      queue_items_hash.each do |queue_item_element|
+        queue_item = queue_items.find_by(id: queue_item_element['id'])
+        next if queue_item.nil?
+        update_order_value queue_item, queue_item_element['order_value']
+        update_rating queue_item, queue_item_element['rating']
+      end
+      normalize_queue_items
+    end
+  end
+
+  private
+
+  def update_order_value(queue_item, order_value)
+    order_value = order_value.to_i
+    if order_value == 0
+      raise ArgumentError.new 'invalid order_value'
+    end
+    queue_item.update_attributes! order_value: order_value if !queue_item.nil?
```

Comment:
`if !` is just `unless` in Ruby. But anyway `if !queue_item.nil?` you could just do `if queue_item` - You should get used that in ruby, everything is truthy except nil and false. So if queue_item is nil then its not going to follow `if` to truthy statement.